### ISSUE TITLE
Prevent Duplicate Project Imports and Display Already Imported Projects

### ISSUE THREAD

**Issue by Ogieeeeee** [microsoft/devhome#2029]
### Suggested new feature or improvement

While it’s possible to continually return to the machine configuration, it would be more efficient if, upon clicking, you could immediately view the projects you’ve imported from GitHub. Currently, the system prevents duplication by failing to add a project if it already exists. This means you only discover a project is a duplicate when you attempt to add it and the action fails. However, it would be beneficial to have a feature that prevents the addition of duplicate projects right from the selection menu, enhancing the user experience by providing this information upfront.

### Scenario

Shoutout to chat gpt ✌️

Imagine you’re a developer named Alex who uses a platform that allows you to import your GitHub projects for further development.

One day, you decide to set up a new machine configuration. You click on the ‘Import from GitHub’ button and a selection menu appears. You see a list of all your GitHub projects and choose a few to import.

A few days later, you return to the machine configuration to add more projects. You click on the ‘Import from GitHub’ button again, but you don’t remember which projects you’ve already imported. You select a project that you think hasn’t been imported yet.

When you try to add it, an error message pops up saying, “This project already exists.” You realize that the project was already imported during your previous session.

You think to yourself, “It would be so much easier if the system could show me which projects I’ve already imported right from this selection menu. That way, I wouldn’t waste time trying to add duplicate projects.”

In this scenario, the platform could improve the user experience by providing information about imported projects upfront, preventing the addition of duplicate projects right from the selection menu. This feature would save users like Alex time and make the process of setting up machine configurations more efficient.

### TL;DR

I can't see the repositories that I have already added in the Machine configuration.

![image](https://github.com/microsoft/devhome/assets/22215187/566d22d4-cea1-4604-9532-a93785f36193)

Also it would be nice to omit the repositories from the selection menu that are already imported

I'm talking about this menu:
![image](https://github.com/microsoft/devhome/assets/22215187/12803d1b-8fb6-4ae9-a15d-2934f4e748f8)

**Comment by krschau** [comment]
Duplicate of #702
